sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -0,0 +1,347 @@
+
+```ts
+/**
+ *  Authenticate with client secret.
+ */
+function withClientSecretCredential() {
+  const credential = new ClientSecretCredential(
+    "<YOUR_TENANT_ID>",
+    "<YOUR_CLIENT_ID>",
+    "<YOUR_CLIENT_SECRET>"
+  );
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a service principal with environment credentials
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `EnvironmentCredential`. The `EnvironmentCredential` looks for well-known environment variable names to determine how it should authenticate. It effectively acts as a wrapper for the `ClientSecretCredential`, `ClientCertificateCredential` or `UsernamePasswordCredential` depending on which environment variables are present.
+
+For more information about how to create a service principal and get these values please refer to [Creating a Service Principal with the Azure CLI][service_principal_azure_cli].
+
+```ts
+/**
+ *  Authenticate with a client certificate.
+ */
+function withEnvironmentCredential() {
+  let credential = new EnvironmentCredential();
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a service principal with a client certificate
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `ClientCertificateCredential`.
+
+For more information about how to create a service principal and get these values please refer to [Creating a Service Principal with the Azure CLI][service_principal_azure_cli].
+
+```ts
+/**
+ *  Authenticate with a client certificate.
+ */
+function withClientCertificateCredential() {
+  let credential = new ClientCertificateCredential(
+    "<YOUR_TENANT_ID>",
+    "<YOUR_CLIENT_ID>",
+    "<PATH_TO_CERTIFICATE>"
+  );
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+}
+```
+
+## Authenticating a user account with device code flow
+
+This example demonstrates authenticating the `SecretClient` from the [@azure/keyvault-secrets][secrets_client_library] client library using the `DeviceCodeCredential`.
+
+For more information about how to configure an AAD application for device code flow please refer to [Enable applications for device code flow][device_code_flow].
+
+> The `DeviceCodeCredential` offers a credential that can be used with little to no setup - the user is free to use whatever browser they choose to complete the authentication process.
+
+```ts
+/**
+ *  Authenticate with a device code.
+ */
+function withDeviceCodeCredential() {
+  let credential = new DeviceCodeCredential(
+    process.env.AZURE_TENANT_ID,
```

Comment:
Found a bug too late! DeviceCodeCredential doesn't accept direct options anymore. All the parameters are in the options bag.

Reply:
Is that change GA? If not- how about making the sample change after the new changes are GA? Thoughts? I'm just wondering if we make the change now and someone sees and tries to use this sample with the latest package that it wont work